### PR TITLE
feat: LT-1758 - Temporary Export Details

### PR DESCRIPTION
**Pull request by iichr**
- Added a new compulsory section for open and standard application with temporary exports only

https://uktrade.atlassian.net/browse/LT-1758

### CODE REVIEW

File: applications/creators.py [hunk 1]
```diff
@@ -190,6 +191,19 @@ def _validate_additional_information(draft, errors):
     return errors
 
 
+def _validate_temporary_export_details(draft, errors):
+    if draft.case_type.sub_type in [CaseTypeSubTypeEnum.STANDARD, CaseTypeSubTypeEnum.OPEN]:
+        if draft.export_type == ApplicationExportType.TEMPORARY:
```
Comment (andytaylorkainos) [inline applications/creators.py]: should these nested `if`s be combined?

File: applications/libraries/edit_applications.py [hunk 2]
```diff
@@ -63,15 +71,54 @@ def _transform_values(old_end_use_value, new_end_use_value):
...
+                if isinstance(new_temp_export_val, date):
+                    audit_trail_service.create(
+                        actor=request.user,
+                        verb=AuditType.UPDATE_APPLICATION_TEMPORARY_EXPORT,
+                        target=application.get_case(),
+                        payload={
+                            "temp_export_detail": TEMP_EXPORT_DETAILS_FIELDS[key],
+                            "old_temp_export_detail": convert_date_to_string(old_temp_export_val)
+                            if old_temp_export_val
+                            else "",
```
Comment (andytaylorkainos) [inline applications/libraries/edit_applications.py]: hard to follow this because its multilined within a dictionary. can you extract the variable out and do the `if-else` there?

File: applications/creators.py (hunk 1 above)
Comment (iichr) [reply applications/creators.py]: yeah sure

File: applications/libraries/edit_applications.py (hunk 2 above)
Comment (iichr) [reply applications/libraries/edit_applications.py]: sorted

File: applications/views/temporary_export_details.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    authentication_classes = (ExporterAuthentication,)
+
+    @authorised_users(ExporterUser)
+    @allowed_application_types([CaseTypeSubTypeEnum.OPEN, CaseTypeSubTypeEnum.STANDARD])
+    @application_in_major_editable_state()
+    def put(self, request, application):
+        serializer = get_temp_export_details_update_serializer(application.export_type)
+        serializer = serializer(application, data=request.data, partial=True)
+
+        if not serializer.is_valid():
```
[the remark this answers is not in the document]
Comment (iichr) [reply applications/views/temporary_export_details.py]: nice, didn't know this existed 👌

File: applications/serializers/temporary_export_details.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+                    "temp_direct_control_details": strings.Generic.TemporaryExportDetails.Error.PRODUCTS_UNDER_DIRECT_CONTROL_MISSING_DETAILS
+                }
+            )
+
+        validated_data = super().validate(data)
+
+        today = timezone.now().date()
+        if "proposed_return_date" in validated_data:
+            if validated_data.get("proposed_return_date"):
+                if validated_data["proposed_return_date"] <= today:
```
Comment (andytaylorkainos) [inline applications/serializers/temporary_export_details.py]: you could wrap the last two if statements into one expression:

```
if validated_data.get("proposed_return_date", 0) <= today:
   ...
```

File: applications/serializers/temporary_export_details.py [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+                raise serializers.ValidationError(
+                    {"proposed_return_date": strings.Generic.TemporaryExportDetails.Error.PROPOSED_RETURN_DATE_BLANK}
+                )
+
+        return validated_data
+
+    def update(self, instance, validated_data):
+        standalone_fields = ["temp_export_details", "proposed_return_date"]
+
+        for field in standalone_fields:
```
Comment (andytaylorkainos) [inline applications/serializers/temporary_export_details.py]: considering its only two fields, could you please use the same approach used with `temp_direct_control_details`?

File: applications/libraries/edit_applications.py [hunk 6]
```diff
@@ -63,15 +71,53 @@ def _transform_values(old_end_use_value, new_end_use_value):
...
+                    audit_trail_service.create(
+                        actor=request.user,
+                        verb=AuditType.UPDATE_APPLICATION_TEMPORARY_EXPORT,
+                        target=application.get_case(),
+                        payload={
+                            "temp_export_detail": TEMP_EXPORT_DETAILS_FIELDS[key],
+                            "old_temp_export_detail": old_temp_export_val,
+                            "new_temp_export_detail": new_temp_export_val,
+                        },
+                    )
```
Comment (andytaylorkainos) [inline applications/libraries/edit_applications.py]: you could extract this whole block
```
audit_trail_service.create(
                        actor=request.user,
                        verb=AuditType.UPDATE_APPLICATION_TEMPORARY_EXPORT,
                        target=application.get_case(),
                        payload={
                            "temp_export_detail": TEMP_EXPORT_DETAILS_FIELDS[key],
                            "old_temp_export_detail": old_date_string,
                            "new_temp_export_detail": convert_date_to_string(new_temp_export_val),
                        },
                    )
```

and put it after the `if-else` (transform the old/new_temp_export_detail `if-else`

File: applications/serializers/temporary_export_details.py (hunk 5 above)
Comment (iichr) [reply applications/serializers/temporary_export_details.py]: Sorted, do have a look
